### PR TITLE
Reimplement bin using `bin_start`, `bin_end`

### PR DESCRIPTION
**Pull request by kanitw**
- Reimplement binning to use linear scale instead of ordinal scale based on @jheer’s example in https://github.com/vega/vega/pull/388 using - use the new bin transform’s `bin_end`

- Make `scale.zero` automatically determined rather than always `true`
  - always returns `false` for bin and time
  - Otherwise, return true for `X`, `Y` and `SIZE`

## TODO
- use `bin_mid` instead

### CODE REVIEW

**Comment by kanitw** [comment]
Okay.  Things looks good now for histograms (and when replacing `bar` with `point` and `line`). 
The `area` one still looks odd, but I am not sure how to make it looks better so I defer optimizing using bin with area for #694 .  

vega-lite_gallery

File: gallery/examples.js [hunk 1]
```diff
@@ -252,6 +253,81 @@ var EXAMPLES = [
       },
       'data': {'url': 'data/cars.json'}
     }
+  },{
```
Comment (domoritz) [inline gallery/examples.js]: I'll put these in separate files some time later.

Comment (kanitw) [inline gallery/examples.js]: I think refering for exemplar ones in vega-editor make sense.  For ones that are more visual test (like histogram with point, line -- we should make sure that it is easy to create ones)

Having it here, reduce the need to name the file nicely (and burden of renaming if the original name sucks, etc.)  That said, this examples.js can become too damn long so I'm not against splitting it.  Just wanna express potential downside of that here.

Comment (domoritz) [inline gallery/examples.js]: I have similar concerns. We will see after next week.

**Pull request merged**